docs/pages/reference/access-controls/roles.mdx

```diff
@@ -0,0 +1,61 @@
+---
+title: Teleport Role Configuration Reference
+description: The detailed guide and reference documentation for Teleport Role YAML Configuration.
+h1: Role YAML Configuration
+---
+
+(!docs/pages/includes/backup-warning.mdx!)
+
+## Example Role YAML
+
+```yaml
+# role.yaml
+
+kind: role
+version: v3
+metadata:
+  name: example-role
+spec:
+(!docs/pages/includes/roles_config.yaml!)
+```
+
+## YAML Table
+
+| Attribute | Parent Field | Required | Description | Default Value | Accepted Values |
+| --- | --- |  --- | --- |  --- | --- |
+| kind | | | | |
+| version | | | | |
+| metadata | | | | |
+| name | metadata | | | | 
+| spec |  | | | | 
+
+## Example Role impersonation YAML
+
+Two roles can be combined into one YAML file. 
+
+The example below shows on role that [impersonates](../../access-controls/guides/impersonation.mdx) the other.
+
+```yaml
+# role-impersonation.yaml
+
+kind: role
+version: v3
+metadata:
+  name: testrole
+spec:
+  allow:
+    logins: ['access']
+---
+kind: role
+version: v3
+metadata:
+  name: imperole
+spec:
+  allow:
+    impersonate:
+      roles: ['testrole']
+```
+
```

Comment:
Add examples of suggested reviewers from dual auth per https://github.com/gravitational/teleport/pull/7421